src/components/User.vue

```diff
@@ -1,38 +1,122 @@
-<script setup lang="ts">
-  import { createApp } from 'vue';
+<script lang="ts">
+import { defineComponent, ref, watchEffect } from 'vue';
+import Search from './Search.vue';
+import InfoBlock from './InfoBlock.vue';
+import Avatar from './Avatar.vue';
+import ProfileInfo from './ProfileInfo.vue';
+import Bio from './Bio.vue';
+import Stats from './Stats.vue';
+import StatItem from './StatItem.vue';
 
+interface UserInfo {
+  avatar_url: string;
+  name: string | null;
+  login: string;
+  created_at: string;
+  bio: string | null;
+  public_repos: number;
+  followers: number;
+  following: number;
+  location: string | null;
+  blog: string | null;
+  twitter_username: string | null;
+  company: string | null;
+}
+
+export default defineComponent({
+  components: {
+    Search,
+    InfoBlock,
+    Avatar,
+    ProfileInfo,
+    Bio,
+    Stats,
+    StatItem,
+  },
```

Comment:
It would be good to add type annotations to the search and error props using the PropType utility provided by Vue. This makes it clear what type of values these props expect and can help prevent errors or confusion when using the component.